x-pack/plugins/secret_service/server/service/secret_service.test.ts

```diff
@@ -0,0 +1,179 @@
+    const hidden = await subject.hideAttribute(hideMe);
+    expect(savedObjectsClient.create).toHaveBeenCalledTimes(1);
+    expect(savedObjectsClient.update).not.toHaveBeenCalled();
+    expect(hidden).toBeDefined();
+    expect(typeof hidden).toBe('string');
+    expect(hidden).toMatch(/^[\w+\/]+/);
+    expect(subject);
+  });
+
+  it('should be able to unhide values from saved objects', async () => {
+    expect.assertions(3);
+    const hideMe = {
+      message: 'my secret message',
+      anotherSecret: 'this is unhidden',
+      someOtherProp: 'check me',
+    };
+    let insideJob: any = null;
+    savedObjectsClient.create.mockImplementation((type: string, attributes: any, options?: any) => {
+      insideJob = {
+        id: options.id,
+        type: 'testSecretType',
+        attributes,
+      };
+      return insideJob;
+    });
+    savedObjectsClient.get.mockImplementation((type: string, id: string, options?: any) => {
+      return insideJob;
+    });
+    const hidden = await subject.hideAttribute(hideMe);
+    const unhidden = await subject.unhideAttribute(hidden);
+    expect(unhidden.id).toEqual(hidden);
+    expect(unhidden.type).toEqual('testSecretType');
+    expect(unhidden.attributes).toMatchObject(hideMe);
+  });
+
+  it('should be unable to unhide values if the saved object was changed in any way', async () => {
+    expect.assertions(1);
+    const hideMe = {
+      message: 'my secret message',
+      anotherSecret: 'this is unhidden',
+      someOtherProp: 'check me',
+    };
+    let insideJob: any = null;
+    const internalSavedObject: any = null;
+    savedObjectsClient.create.mockImplementation((type: string, attributes: any, options?: any) => {
+      insideJob = {
+        id: 'testId',
+        type: 'testSecretType',
+        attributes,
+      };
+      return insideJob;
+    });
+    savedObjectsClient.get.mockImplementation((type: string, id: string, options?: any) => {
+      internalSavedObject.anotherSecret = 'I am changing the message';
+      internalSavedObject.addNewField = 'me';
+      return internalSavedObject;
+    });
+    const hidden = await subject.hideAttribute(hideMe);
+
+    try {
+      await subject.unhideAttribute(hidden);
+      fail('The test fails because no exception was thrown');
+    } catch (e) {
+      expect(e).not.toBeNull();
+    }
+  });
+
+  describe('encryption key validation', async () => {
+    it('should be valid if first time use', async () => {
+      let dummyObj: any;
+      savedObjectsClient.create.mockImplementation(
+        (type: string, attributes: any, options?: any) => {
+          dummyObj = {
+            id: options.id,
+            type,
+            attributes,
+          };
+          return dummyObj;
+        }
+      );
+      savedObjectsClient.get.mockImplementation((type: string, attributes: any, options?: any) => {
+        return dummyObj;
+      });
+      const isValid = await subject.validateKey();
+      expect(isValid).toBeTruthy();
```

Comment:
__optional nit:__ `expect(isValid).toBe(true);` feels a bit more correct (and similar note for test cases below).